contrib/dist/linux/buildrpm.sh

```diff
@@ -267,24 +276,30 @@ fi
 # Find where the top RPM-building directory is
 #
 
-file=~/.rpmmacros
-if test -r $file; then
-    rpmtopdir=${rpmtopdir:-"`grep %_topdir $file | awk '{ print $2 }'`"}
+# if the user did not specify an $rpmtopdir, check for an .rpmmacros file.
+if test "$rpmtopdir" == ""; then
+    file=~/.rpmmacros
+    if test -r $file; then
+        rpmtopdir=${rpmtopdir:-"`grep %_topdir $file | awk '{ print $2 }'`"}
+    fi
 fi
+
+# If needed, initialize the $rpmtopdir directory. If no $rpmtopdir was
+# specified, try various system-level defaults.
 if test "$rpmtopdir" != ""; then
-	rpmbuild_options="$rpmbuild_options --define '_topdir $rpmtopdir'"
+    rpmbuild_options="$rpmbuild_options --define '_topdir $rpmtopdir'"
```

Comment:
Minor quibble: it would be nice to split this commit into two commits:
1. One that makes the desired `-R` code change
1. Another that makes the whitespace fixes (and no other code/logic changes).

Reply:
I can do that.

Reply:
Done!